proposals/assign-operator-overload.md

```diff
@@ -0,0 +1,441 @@
+
+
+## Motivation
+
+Let’s say you have mutable value container classes, for example defined as:
+
+
+```kotlin
+class Property<T>(var value: T) {
+      /** Get the current value of this property. */
+      fun get(): String = value
+      /** Set the value of this property to the given value eagerly. */
+      fun set(value: T) {
+          this.value = value
+      }
+      /** Make this property mirror the value of the given source. */
+      fun set(source: Property<T>) {
+         this.value = source.get()
+      }
+}
+```
+
+And you use such container class on your object:
+
+```kotlin
+data class MyClass(val prop: Property<String> = Property<String>(""))
+```
+
+Then you have to modify `prop` by explicitly calling `.set`. What we propose is to be able to overload assign operator `=` so one could set a container value with `=`. That would simplify use cases especially for DSL scripts. 
+
+
+### Example
+
+In Gradle, users construct tasks as classes that have _inputs_ or _outputs_ modeled as container objects. That gives Gradle the benefit of “tracking" properties and with that option to handle the lifecycle of a property, detect dependencies between tasks etc. Additionally users can also specify values lazily. Lets show a simplified version of Gradle tasks.
+
+
+Let’s say that we have a class like, where Property is a class defined in the [motivation section](#motivation):
+```kotlin
+open class MyTask {
+    val input1: Property<File> = Property<File>()
+    val input2: Property<File> = Property<File>()
+    val output: Property<File> = Property<File>()
+}
+```
+
+Given an instance of MyTask, you'd set values like:
+
+```kotlin
+val myTask = MyTask()
+val calculatedProperty: Property<File> = calculateProperty()
+myTask.input1.set(File("input-file.txt"))
+myTask.input2.set(calculatedProperty)
+myTask.output.set(File("input-file.txt"))
+```
+
+Or as a DSL
+
+```
+myTask.apply {
+    input1.set(File("input-file.txt"))
+    input2.set(calculatedProperty)
+    output.set(File("input-file.txt"))
+}
+```
+
+This can soon become very verbose, especially if properties are set with complex expressions.
+With the option to overload the `=` operator one could write tasks like:
+
+```kotlin
+val myTask = MyTask()
+val calculatedProperty: Property<File> = calculateProperty()
+myTask.input1 = File("input-file.txt")
+myTask.input2 = calculatedProperty
+myTask.output = File("input-file.txt")
+```
+
+Or as a DSL
+
+```kotlin
+myTask.apply {
+    input1 = File("input-file.txt")
+    Input2 = calculatedProperty
```

Comment:
`Input2` is accidentally capitalized here.
There are also a couple more places in the document where this typo has leaked.

Reply:
Thank you, should be fixed now.